### PR TITLE
do not emit exportsStar function if module does not expose any values

### PR DESCRIPTION
**Pull request by vladima**
Tentative fix proposal for #4492. I'm a bit concerned about perf implications in case of deep chaining of `export *`: if list of exports was not realized during compilation (i.e. because it is public API that should be used by external consumers) then calling `getExportsOfModule` during emit will trigger cloning of symbol tables just to check the if one of symbols is value. I'll collect some perf data on Angular codebase to see if perf impact is noticeable.

### CODE REVIEW

**Comment by vladima** [comment]
Don't see noticeable impact when compiling angular codebase

**Comment by sandersn** [comment]
Looks good to me, but I know little about the emitter so probably somebody else should OK the merge too.

**Comment by vladima** [comment]
pinging @mhegazy can you check if this PR is ok?

**Comment by vladima** [comment]
@mhegazy FYI, this are results of perf test run on Angular2 codebase

```
// master
Parse:	min: 1.030000	max: 1.050000	average 1.040000	variance: 0.000040	stddev: 0.006325
Bind:	min: 0.520000	max: 0.540000	average 0.524000	variance: 0.000064	stddev: 0.008000
Check:	min: 3.030000	max: 3.170000	average 3.112000	variance: 0.003536	stddev: 0.059464
Emit:	min: 1.860000	max: 2.350000	average 2.070000	variance: 0.039120	stddev: 0.197788
Total:	min: 6.590000	max: 7.070000	average 6.742000	variance: 0.038656	stddev: 0.196611

// current branch
Parse:	min: 1.020000	max: 1.060000	average 1.036000	variance: 0.000184	stddev: 0.013565
Bind:	min: 0.510000	max: 0.530000	average 0.520000	variance: 0.000080	stddev: 0.008944
Check:	min: 3.000000	max: 3.130000	average 3.096000	variance: 0.002544	stddev: 0.050438
Emit:	min: 1.870000	max: 2.290000	average 2.096000	variance: 0.027024	stddev: 0.164390
Total:	min: 6.530000	max: 6.950000	average 6.748000	variance: 0.031296	stddev: 0.176907

```

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -14672,6 +14672,33 @@ namespace ts {
...
 
+        function moduleExportsSomeValue(moduleReferenceExpression: Expression): boolean {
+            let moduleSymbol = resolveExternalModuleName(moduleReferenceExpression.parent, moduleReferenceExpression);
+            if (!moduleSymbol) {
+                // module not found - be conservative
+                return true;
+            }
+
+            const hasExportAssignment = getExportAssignmentSymbol(moduleSymbol) !== undefined;
+            moduleSymbol = resolveExternalModuleSymbol(moduleSymbol);
```
Comment (mhegazy) [inline src/compiler/checker.ts]: if you can add a comment here, it would be great.

**Pull request merged**